cmd/treedb_text_ingest_qual/contract.go

```
@@ -0,0 +1,596 @@
+		validScale = validScale || r.Scale == s
+	}
+	if !validScale || r.Repetition < 1 {
+		return fmt.Errorf("invalid scale or repetition")
+	}
+	if r.PeakRSSScope != "fresh_process_per_mode" || r.PeakRSSPID < 1 {
+		return fmt.Errorf("peak RSS requires a fresh process measurement")
+	}
+	if r.SourceDocuments != r.Scale || r.GeneratedChunks < 0 || r.IndexedLiveRows < 1 || r.IndexedParentRows < 0 {
+		return fmt.Errorf("document accounting is incomplete")
+	}
+	if r.Mode == "source_chunk" && (r.GeneratedChunks < 1 || !r.ParentsTextIndexed || r.IndexedParentRows != r.SourceDocuments || r.IndexedLiveRows != r.IndexedParentRows+r.GeneratedChunks || r.ChunkBatchSize != min(sourceChunkBatchLimit, r.SourceDocuments) || r.ChunkBatchCount != (r.SourceDocuments+sourceChunkBatchLimit-1)/sourceChunkBatchLimit || r.Generations != uint64(r.ChunkBatchCount+1)) {
```

golangci-lint (2.12.2)

[high] G115: integer overflow conversion int -> uint64

(gosec)